### PR TITLE
Type annotations

### PR DESCRIPTION
**Pull request by yoonieaj**
## Proposed Changes

Added type interfaces and type annotations for functions and constants in `style.ts`. This should improve code organisation/readability.
...

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |          |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |          |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |     x    |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |          |
| 📚 _Documentation update_ (change that _only_ updates documentation)                    |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

-   [x] I have performed a self-review of my changes.
    -   Check that all changed files included in this pull request are intentional changes.
    -   Check that all changes are relevant to the purpose of this pull request, as described above.
-   [ ] I have added tests for my changes, if applicable.
    -   This is **required** for all bug fixes and new features.
-   [ ] I have updated the project documentation, if applicable.
    -   This is **required** for new features.
-   [ ] If this is my first contribution, I have added myself to the list of contributors.
-   [x] I have updated the project Changelog (this is required for all changes).

After opening your pull request:

-   [x] I have verified that the CI tests have passed.
-   [x] I have reviewed the test coverage changes reported by Coveralls.
-   [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

Hello Professor Liu! I have a few questions about the changes I've made so far:
1. The code doesn't seem to care whether the DrawnObject attributes are optional, so I'm not sure if I made the right call on which attributes should be optional?
2. Since the preset Styles only specify certain attributes, can I leave every one of these attributes as optional?

### CODE REVIEW

**Comment by david-yz-liu** [comment]
1. Weird that the compiler doesn't seem to be catching that, but we can look into that later. I think `show_indexes` and `style` can both be optional as well.
2. Yes it's good for all `Style` attributes to be optional 👍

**Comment by yoonieaj** [comment]
## Proposed Changes

Added type interfaces and type annotations for functions and constants in `style.ts`. This should improve code organisation/readability.
...

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |          |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |          |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |     x    |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |          |
| 📚 _Documentation update_ (change that _only_ updates documentation)                    |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

-   [x] I have performed a self-review of my changes.
    -   Check that all changed files included in this pull request are intentional changes.
    -   Check that all changes are relevant to the purpose of this pull request, as described above.
-   [ ] I have added tests for my changes, if applicable.
    -   This is **required** for all bug fixes and new features.
-   [ ] I have updated the project documentation, if applicable.
    -   This is **required** for new features.
-   [ ] If this is my first contribution, I have added myself to the list of contributors.
-   [x] I have updated the project Changelog (this is required for all changes).

After opening your pull request:

-   [ ] I have verified that the CI tests have passed.
-   [ ] I have reviewed the test coverage changes reported by Coveralls.
-   [ ] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

File: memory-viz/src/automate.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ function drawAutomatedStackFrames(stack_frames, configuration) {
         let width;
         let height;
 
-        if (stack_frame.name !== "BLANK") {
+        if (stack_frame.type !== "BLANK") {
```
Comment (david-yz-liu) [inline memory-viz/src/automate.ts:76]: This change is out of scope of this PR (and in fact I suspect it's related to what @sarahsonder is working on currently).

File: memory-viz/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+interface DrawnObject {
```
Comment (david-yz-liu) [inline memory-viz/src/types.ts:1]: Use the `export` keyword for these types, and then import them explicitly in `style.ts`.

Comment (david-yz-liu) [reply memory-viz/src/types.ts:1]: Also rename this type to `DrawnEntity`, since the word "object" has another meaning in the context of memory model diagrams.

**Review by david-yz-liu — COMMENTED** [review]
@yoonieaj great work. I left a few inline comments. Also:

1. I see that you left a comment with your full PR description. Instead, you can *edit* your existing PR description and then copy back in the contents of that comment.
2. Since you've defined the `DrawnObject` type very well, please go ahead and modify the other source code files to use this type annotation instead of the old "object" type. You _don't_ need to add type annotations to all functions/values in those other files though, just make sure to use `DrawnObject` where appropriate.

File: memory-viz/src/types.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+export interface DrawnEntity {
+    name?: string;
+    type?: string;
+    x?: number;
+    y?: number;
+    id: number | string;
+    value: any;
+    show_indexes?: boolean;
+    style?: Style;
+    height: number;
```
Comment (david-yz-liu) [inline memory-viz/src/types.ts:10]: I think this, `width`, and `rowBreaker` are also optional?

File: memory-viz/src/automate.ts [hunk 4]
```diff
@@ -244,7 +245,7 @@ function drawAutomatedOtherItems(
  * The returned object has two attributes as 'stack_frames' and 'other_items'.
  * Each of these attributes are a list of objects that were originally given by the user.
  *
- * @param {object[]} objects - The list of objects, including stack-frames (if any) and other items, that
+ * @param {Object[]} objects - The list of objects, including stack-frames (if any) and other items, that
```
Comment (david-yz-liu) [inline memory-viz/src/automate.ts:248]: Is this meant to be `DrawnEntity`?

**Review by david-yz-liu — APPROVED** [review]
